Assets/Scripts/Tools/Extrude.cs

```diff
@@ -222,6 +243,164 @@ public void OnTriggerEnter(Collider other)
         }
     }
 
+    // Actually create the vertex and edge GameObject interactables
+    void CreateVisuals(MeshRebuilder meshRebuilder, List<Vector3> newVertices, List<int> newTriangles, int oldLengthVert, int oldLengthTri)
+    {
+        Vector3[]vertices = meshRebuilder.vertices;
+        int[] triangles = meshRebuilder.triangles;
+
+        for (int i = 0; i < vertices.Length; i++)
+        {
+            // Create a new vertex from a prefab, make it a child of the mesh and set it's position
+            GameObject newVertex = Instantiate(vertex, meshRebuilder.model.transform);
+            
+            newVertex.transform.localPosition = vertices[i];
+
+            newVertex.name = "Vertex" + i.ToString();
+
+            // Set the id of the Vertex component to be the index in the vertices array
+            Vertex vertexObj = newVertex.GetComponent<Vertex>();
+            vertexObj.id = i;
+            meshRebuilder.vertexObjects.Add(vertexObj);
+
+            if (!ToolManager.instance.grabVertex)
+            {
+                newVertex.SetActive(false);
+            }
+
+            // Save vertices adjacent to the one we're currently looking at (no duplicates)
+            HashSet<int> adjacentVertices = new HashSet<int>();
+
+            // Loop through the triangles array and look for the adjacent vertices
+            for (int j = 0; j < triangles.Length; j += 3)
+            {
+                // Triangles are created in triplets
+                // Entering "0, 1, 2," in the triangles array would make a triangle
+
+                if (triangles[j] == i) // First index of triplet
+                {
+                    adjacentVertices.Add(triangles[j + 1]);
+                    adjacentVertices.Add(triangles[j + 2]);
+                }
+                else if (triangles[j + 1] == i) // Second index of triplet
+                {
+                    adjacentVertices.Add(triangles[j]);
+                    adjacentVertices.Add(triangles[j + 2]);
+                }
+                else if (triangles[j + 2] == i) // Third index of triplet
+                {
+                    adjacentVertices.Add(triangles[j]);
+                    adjacentVertices.Add(triangles[j + 1]);
+                }
+            }
+
+            // Connect a line from our starting vertex to each adjacent vertex
+            foreach (int k in adjacentVertices)
+            {
+                // Ignore adjacent vertices we've already dealt with
+                if (k < i)
+                    continue;
+
+                // Same as vertex, create a new edge object and set its parent
+                GameObject newEdge = Instantiate(edge, meshRebuilder.model.transform);
+                Component[] components = newEdge.GetComponents(typeof(Component));
```

Comment:
Is this needed? If not, we can probably remove the logging and GetComponents() method

Reply:
I tested removing this and I think this was the culprit for the worse performance actually. I'll push a commit to remove it if we don't need it.